flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesResourceManagerDriver.java

```diff
@@ -85,23 +86,31 @@
 
 	public KubernetesResourceManagerDriver(
 			Configuration flinkConfig,
-			FlinkKubeClient kubeClient,
 			KubernetesResourceManagerDriverConfiguration configuration) {
 		super(flinkConfig, GlobalConfiguration.loadConfiguration());
-
 		this.clusterId = Preconditions.checkNotNull(configuration.getClusterId());
 		this.podCreationRetryInterval = Preconditions.checkNotNull(configuration.getPodCreationRetryInterval());
-		this.kubeClient = Preconditions.checkNotNull(kubeClient);
 		this.requestResourceFutures = new HashMap<>();
 		this.podCreationCoolDown = FutureUtils.completedVoidFuture();
 	}
 
+	public KubernetesResourceManagerDriver(
+			Configuration flinkConfig,
+			FlinkKubeClient kubeClient,
+			KubernetesResourceManagerDriverConfiguration configuration) {
+		this(flinkConfig, configuration);
+		this.kubeClient = Preconditions.checkNotNull(kubeClient);
+	}
```

Comment:
Instead of introducing another constructor, I think we can have only one constructor that takes `KubeClientFactory` as an argument. We would also need to make `KubeClientFactory` and interface and provide a singleton implementation.
In this way, the tests can provide a testing factory implementation, without needing to have a separate code path in the production codes.